addon/v1alpha1/types_clustermanagementaddon.go

```diff
@@ -175,10 +185,146 @@ type PlacementStrategy struct {
+	//   reference for finding and checking canary clusters before upgrading all. To trigger the rollout
+	//   on the canary clusters, you can define another rollout strategy with the type RollingUpdate, or even
+	//   manually upgrade the addons on those clusters.
+	//
+	// +kubebuilder:validation:Enum=UpdateAll;RollingUpdate;RollingUpdateWithCanary
+	// +kubebuilder:default:=UpdateAll
+	// +optional
+	Type string `json:"type"`
+
+	// Rolling update with placement config params. Present only if the type is RollingUpdate.
+	// +optional
+	RollingUpdate *RollingUpdate `json:"rollingUpdate,omitempty"`
+
+	// Rolling update with placement config params. Present only if the type is RollingUpdateWithCanary.
+	// +optional
+	RollingUpdateWithCanary *RollingUpdateWithCanary `json:"rollingUpdateWithCanary,omitempty"`
+}
+
+const (
+	// AddonRolloutStrategyUpdateAll is the addon rollout strategy representing apply the new configs to
+	// all the selected clusters at once.
+	AddonRolloutStrategyUpdateAll string = "UpdateAll"
+	// AddonRolloutStrategyRollingUpdate is the addon rollout strategy representing apply the new configs to
+	// all the selected clusters with the concurrency rate.
+	AddonRolloutStrategyRollingUpdate string = "RollingUpdate"
+	// AddonRolloutStrategyRollingUpdate is the addon rollout strategy representing wait and check
+	// if add-ons on the canary have applied the new configs, then apply the new configs to
+	// all the selected clusters with the concurrency rate.
+	AddonRolloutStrategyRollingUpdateWithCanary string = "RollingUpdateWithCanary"
+)
+
+// RollingUpdate represents the behavior to rolling update add-on configurations
+// on the selected clusters.
+type RollingUpdate struct {
+	// The maximum concurrently updating number of addons.
+	// Value can be an absolute number (ex: 5) or a percentage of desired addons (ex: 10%).
+	// Absolute number is calculated from percentage by rounding up.
+	// Defaults to 25%.
+	// Example: when this is set to 30%, once the addon configs change, the addon on 30% of the selected clusters
+	// will adopt the new configs. When the addons with new configs are healthy, the addon on the remaining clusters
+	// will be further updated.
+	// +kubebuilder:default:="25%"
+	// +optional
+	MaxConcurrency intstr.IntOrString `json:"maxConcurrency,omitempty"`
+}
+
+// RollingUpdateWithCanary represents the canary placement and behavior to rolling update add-on configurations
+// on the selected clusters.
+type RollingUpdateWithCanary struct {
+	// Canary placement reference.
+	// +kubebuilder:validation:Required
+	// +required
+	Placement PlacementRef `json:"placement,omitempty"`
+
+	// the behavior to rolling update add-on configurations.
+	RollingUpdate `json:",inline"`
 }
 
 // ClusterManagementAddOnStatus represents the current status of cluster management add-on.
 type ClusterManagementAddOnStatus struct {
+	// defaultconfigReferences is a list of current add-on default configuration references.
```

Comment:
Are these the configurations that will be applied if a new managedcluster appears in the placement?

Reply:
yes, the ClusterManagementAddOnStatus records the configurations with spec hash, a new added managedcluster will use these configs.